consensus/polybft/state_sync_manager.go

```diff
@@ -82,7 +84,7 @@ func NewStateSyncManager(logger hclog.Logger, state *State, config *stateSyncCon
 		logger: logger.Named("state-sync"),
 		state:  state,
 		config: config,
-		lock:   sync.Mutex{},
+		lock:   sync.RWMutex{},
```

Comment:
Nitpick, mutexes doesn't need to be initialized, but as you wish :slightly_smiling_face: 
```suggestion
```

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1083/commits/37d13b5aeda743254f04235b861ccd87e76e247f